wgpu-core/src/command/mod.rs

```diff
@@ -96,49 +102,107 @@ pub struct BakedCommands<A: HalApi> {
 pub(crate) struct DestroyedBufferError(pub id::BufferId);
 pub(crate) struct DestroyedTextureError(pub id::TextureId);
 
-pub struct CommandBuffer<A: HalApi> {
+pub struct CommandBufferMutable<A: HalApi> {
     encoder: CommandEncoder<A>,
     status: CommandEncoderStatus,
-    pub(crate) device_id: Stored<id::DeviceId>,
     pub(crate) trackers: Tracker<A>,
     buffer_memory_init_actions: Vec<BufferInitTrackerAction>,
     texture_memory_actions: CommandBufferTextureMemoryActions,
-    limits: wgt::Limits,
-    support_clear_texture: bool,
     #[cfg(feature = "trace")]
     pub(crate) commands: Option<Vec<TraceCommand>>,
 }
 
+impl<A: HalApi> CommandBufferMutable<A> {
+    pub(crate) fn open_encoder_and_tracker(&mut self) -> (&mut A::CommandEncoder, &mut Tracker<A>) {
+        let encoder = self.encoder.open();
+        let tracker = &mut self.trackers;
+        (encoder, tracker)
+    }
+    pub(crate) fn raw_mut(
+        &mut self,
+    ) -> (
+        &mut CommandEncoder<A>,
+        &mut CommandEncoderStatus,
+        &mut Tracker<A>,
+        &mut Vec<BufferInitTrackerAction>,
+        &mut CommandBufferTextureMemoryActions,
+    ) {
```

Comment:
What's the point of this function compared to just using the members directly?

Reply:
Oh god! Tnx you spotted it - it was an iteration progress that I forgot to remove